### PR TITLE
Doc changelog

### LINKED ISSUE #105: Feature: Add changelog to docs
## Background and motivation

It looks like the changelog is associated with our newest tagged release, but I'm not sure it is easily discoverable outside of the github interface. It would be great if it showed up more clearly in our docs. 

## Description of idea

We should have a separate, easily discoverable changelog for this project that's included in our documentation. 

## Implementation details
I suggest making a separate `changelog.rst` file, and then make sure it is required that we update it when we put out a new tagged release. the changelog can then be sourced in the `index.rst` file of the docs. [by @munkm]

Checkboxes for features: 
- [x] Release notes template
- [x] Workflow to create new draft release, milestone, and release notes from the template that are all 1 minor version higher than the current release upon publishing of current release
- [x] Workflow that we can manually run to do the same as above but for micro version releases (i.e. patches and hotfixes)
- [x] Workflow to populate the draft release description with the associated release notes on that milestone when those release notes are updated
- [x] Add checkbox to the PR template that requires updating the release notes for the current milestone
- [x] Detail the new parts of the workflow in the Contributing.md document
- [x] Start a dev guide in the docs that details this as well.

The should all be in the same PR
[by @yardasol]

## Potential snags

I don't see any issues. We can have a running changelog file that gets added to every release cycle. [by @munkm]

### PR DESCRIPTION
**Pull request by yardasol**
## Summary of changes

This PR adds several features related to change logs/release notes. Specifically, this PR adds:
- A new section in the docs, `releasenotes`
- A template for writing release notes 
- Workflows that:
  - Auto-populate the current draft release body with the associated release notes whenever those release notes are updated
  - Create a new milestone, draft release, and release notes one minor version higher than the current draft release when the current draft release gets published
  - ~~Do the same as above but for micro versions (manual trigger only)~~
- New content to `CONTRIBUTING.md` describing the release notes requirements
- A new `devguide` section in the docs, consisting of
  - The landing page
  - Conversion of `CONTRIBUTING.md` to `.rst` 
- A new checkbox in the PR template that requires users to document their changes in the release notes for the current milestone.
- Acknowledgement of OpenMC's doc structure and the use of sphinx-multiversion to make versioned documentation.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Required for Merging
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.
- [ ] My change is a source code change
  - [ ] I have added/modified tests to cover my changes
  - [ ] I have explained why my change does not require new/modified tests 
- [x] All new and existing tests passed.
  - [x] CI tests pass
  - [x] Local tests pass (including Serpent2 integration tests)

## Associated Issues and PRs

- Issue: closes #105 

## Associated Developers

- Dev: @munkm 

## Checklist for Reviewers

Reviewers should use [this link](https://arfc.github.io/manual/guides/pull_requests) to get to the 
Review Checklist before they begin their review.

### CODE REVIEW

**Comment by yardasol** [comment]
Do not merge until #98 is merged

File: .github/workflows/deploy-docs.yml
Comment (abachma2) [inline .github/workflows/deploy-docs.yml:9]: Does this remove pushing the updates to the master branch? Or does this do something else?

File: doc/devguide/contributing.rst
Comment (abachma2) [inline doc/devguide/contributing.rst:5]: Is there a reason the information from the `CONTRIBUTING.md` file is repeated here?

File: saltproc/version.py
Comment (abachma2) [inline saltproc/version.py:7]: Why do you have a 0 here? Based on the comment, shouldn't it be `''`?

**Review by abachma2 — COMMENTED** [review]
Overall, this looks good. Could you please explain a little of the github workflow? When to the `next-release-micro` and `next-release-minor` get performed? Do they need to be manually called or do they get triggered by a PR merge? Also do you need a `next-release-major`?

File: .github/workflows/deploy-docs.yml
Comment (yardasol) [reply .github/workflows/deploy-docs.yml:9]: That's exactly what it does!

File: saltproc/version.py
Comment (yardasol) [reply saltproc/version.py:7]: I think I tried that once and it didn't work. I can check though

File: .github/workflows/deploy-docs.yml
Comment (abachma2) [reply .github/workflows/deploy-docs.yml:9]: Is there any reason you are removing this? Does pushing to the master branch trigger this workflow?

File: doc/devguide/contributing.rst
Comment (abachma2) [reply doc/devguide/contributing.rst:5]: So this is what ends up on the webpage separate from the repo?

Comment (yardasol) [reply doc/devguide/contributing.rst:5]: Correct

Comment (yardasol) [reply doc/devguide/contributing.rst:5]: I used pandoc to generate the .rst file from the .md file

File: .github/workflows/deploy-docs.yml
Comment (yardasol) [reply .github/workflows/deploy-docs.yml:9]: I'm removing this because we only need to trigger the workflow when there are changes to the source code (which is what the API documentation is based on) or when there are new doc pages (in the `doc` folder). We don't need to trigger this workflow when we add/edit workflows or configuration files.

**Comment by yardasol** [comment]
@abachma2 to answer your question about `next-version-major.yml`, at some point yes we will want one, but we have not hit a major release version yet. It will have the same structure at `next-version-micro.yml`.

File: saltproc/version.py
Comment (yardasol) [reply saltproc/version.py:7]: I read over this again and the reason we have a 0 there is because we are also using the `_version_extra` to signify that this version is in development and not a release version.

**Comment by yardasol** [comment]
In my meeting with @munkm this morning, we discussed removing the `next_release_micro.yml` workflow as it doesn't really make sense to have with our current release model. It is technically cruft, so I'll be removing it from this PR and putting it in a new feature branch, `branched-releases`, where we can more rigorously implement the needed workflow as discussed in #116.

File: saltproc/version.py
Comment (abachma2) [reply saltproc/version.py:7]: Would you mind changing the comment to reflect this?

Comment (yardasol) [reply saltproc/version.py:7]: I read this over again and now I agree with you

File: .github/workflows/next-release-minor.yml
Comment (yardasol) [inline .github/workflows/next-release-minor.yml:72]:
```suggestion
          sed -i "s/^_version_extra = ''# _version_extra = ''/g" saltproc/version.py
```

**Review by abachma2 — APPROVED** [review]
Looks good!

**Pull request merged**
